server/routes/recipes.js

```diff
@@ -0,0 +1,87 @@
+const router = require('express').Router();
+const mongoose = require('mongoose');
+
+let Recipe = require('../models/recipe.js')
+
+router.route('/').get((req, res) => {
+  return res.send('test');
+});
+
+//Create Recipe
+router.route('/new_recipe').post((req, res) => {
+
+  const _id = new mongoose.Types.ObjectId;
+  const name = req.body.name;
+  const ingredients = req.body.ingredients;
+  const description = req.body.description;
+  const price = req.body.price;
+
+
+  //Checks for
+  if(name.length == 0){
+  	res.send("Recipe must have a name");
+  }
+  if(ingredients.length == 0){
+  	res.send("Must list ingredients")
+  }
+  if(description.length == 0){
+    res.send("Recipe must have a description");
+  }
+  /*if(chef_id.length == 0){
+  	res.send("Must include chef_id");
+  }*/
+  if(price.length == 0){
+    res.send("Recipe must have a price");
+  }
+
+  //Creates recipe
+  const newRecipe = new Recipe( {_id, name, ingredients, description, price} );
+
+   //Saves recipe
+   newRecipe.save()
+  .then( () => res.status(201).json('Recipe Created'))
+  .catch(err => res.status(400).json('Error: '+ err));
+
+});
+
+//Retrieve Recipe by id
+router.route('/get_recipe/:id').get((req, res) => {
+	const _id = req.params.id;
+	Recipe.findById( _id ).lean().exec((err, recipe) => {
+    if(err) {return res.send(err)};
+    return res.end(JSON.stringify(recipe))
+  });
+});
+
+//Delete Recipe by id
+router.route('/delete_recipe/:id').get((req, res) => {
+  console.log('Request body' + req.body);
+	const _id = req.params.id;
+	Recipe.findByIdAndDelete( _id, (err, result) => {
+    if(err) {return res.send(err)};
+    return res.send(result)
+    });
+});
+
+//Update Recipe
+router.route('/update_recipe/:id').post((req, res) => {
+  const _id = req.params.id;
+  console.log('Request body' + req.body);
+	Recipe.findByIdAndUpdate( _id, req.body, { 'new': true })
+    .exec((err, result) => {
+    if(err) {return res.send(err)};
+    return res.send(result)
+  });
+});
+
+//Find recipes by chef
+router.route('/get_recipes_by_chef/:chef_id').get((req, res) => {
+	Recipe.find()
+    .where('chef').equals(req.params.chef_id)
+    .exec((err, result) => {
+      if(err) {return res.send(err)};
+      return res.end(JSON.stringify(recipe))
+  });
+});
```

Comment:
typically the REST way of handling routes is to do something like this:
https://i0.wp.com/chelseatroy.com/wp-content/uploads/2018/07/rest_chart.png?w=609&ssl=1

so in our case it would be:
- `(/recipe).post`  to make a new recipe
- `(/recipe).delete`  to delete a recipe
- `(/recipe/:id).get`  to get a recipe by id
- `(/recipe/chefs/:id).get` or `(/chefs/:id/recipes).get` or `(/users/:id/recipes).get`  to get all recipes by a chef. This depends on how chefs are structured